### PR TITLE
[drci] Remove failed workflows if there is a succeeding one

### PR DESCRIPTION
**Pull request by clee2000**
`workflow_run.workflow_id` seems to uniquely identify workflow jobs based on their files
Use the above to tell that `.github/workflows/trunk.yml` is a failed version of `trunk`
Based on this, take the most recent workflow (based on workflow_run.id) and its corresponding jobs to be the truth

Example
image

### CODE REVIEW

File: torchci/rockset/commons/__sql/recent_pr_workflows_query.sql [hunk 1]
```diff
@@ -21,6 +21,7 @@ WITH recent_shas AS (
 )
 SELECT
   w.id AS workflowId,
+  w.workflow_id as workflowUniqueId,
```
Comment (huydhn) [inline torchci/rockset/commons/__sql/recent_pr_workflows_query.sql:24]: I think having some explanation here is needed as there are yet another ID here to complicate things.  Here is what I get so far:

1. For job, `workflowId` is the workflow ID and `id` is the job ID
2. For workflow, `workflowId` is null, and `id` is now the workflow ID

A bit complicated, but that's ok.  Now there is `workflowUniqueId` from `w.workflow_id` :)

Comment (clee2000) [reply torchci/rockset/commons/__sql/recent_pr_workflows_query.sql:24]: I put it in the types file, does it also need to be in the sql file?

Comment (huydhn) [reply torchci/rockset/commons/__sql/recent_pr_workflows_query.sql:24]: Ah ok, that will do IMO

File: torchci/pages/api/drci/drci.ts [hunk 2]
```diff
@@ -700,27 +701,52 @@ export function reorganizeWorkflows(
...
+      if (!existingWorkflowId || existingWorkflowId! < workflow.id!) {
+        recentWorkflows.set(workflowUniqueId, workflow);
+      }
+    }
+
     // Remove retries
     const removeRetries = new Map();
-    for (const job of prInfo.jobs) {
+    for (const job of jobs) {
+      if (
```
Comment (huydhn) [inline torchci/pages/api/drci/drci.ts:723]: I wonder if this could also get rid of cancelled job, let's see

**Review by huydhn — APPROVED** [review]
LGTM!

**Pull request merged**